src/shared/lib/utils/navigation.ts

```diff
@@ -0,0 +1,13 @@
+import { NavigationProp } from '@react-navigation/native';
+
+export const isActivityExecuting = (
+  navigator: NavigationProp<ReactNavigation.RootParamList>,
+): boolean => {
+  const navigationState = navigator.getState();
+  if (!navigationState) {
+    return false;
+  }
+  const length = navigationState.routes.length;
+  const lastRoute = navigationState.routes[length - 1];
+  return lastRoute.name === 'InProgressActivity';
+};
```

Comment:
Please move the hook to either the activity entity as it is related to business logic

Reply:
I've moved in to shared as separated function because it also used from 2 places